dsc/workflows/archivesspace.py

```diff
@@ -58,14 +59,26 @@ def workflow_specific_processing(self) -> None:
             )
             return
 
+        # create dataframe
+        df = pd.DataFrame(
+            [
+                {
+                    "ao_uri": item.source_system_identifier,
+                    "dspace_handle": (
+                        item.dspace_handle
+                        if item.dspace_handle
+                        else "DSpace handle not set, possible error"
+                    ),
+                }
+                for item in successful_item_submissions
+            ]
+        )
+
         # create report in S3 bucket
         with smart_open.open(
             f"s3://{self.output_path}/{self.batch_id}-{run_date_str}.csv", "w"
         ) as csv_file:
-            writer = csv.writer(csv_file)
-            writer.writerow(["ao_uri", "dspace_handle"])
-            for ao_uri, dspace_handle in handle_uri_mapping.items():
-                writer.writerow([ao_uri, dspace_handle])
+            df.to_csv(csv_file, index=False)
```

Comment:
Awesome!  I forget that we can use `smart_open` to get a file-like object and then pass that to `df.to_csv()`.  Very nice!